docs/NNNN-hash-and-id-n-tuple-storage-layout.md

```diff
@@ -0,0 +1,323 @@
+`tupleSize` determines the number of digest characters to include in each tuple. The tuples are used as directory names. The default value is `3`, which means that each directory in the OCFL storage hierarchy could contain up to 4096 files. Increasing this value increases the maximum number of files per directory.
+
+If `tupleSize` is set to `0`, then no tuples are created and `numberOfTuples` MUST also equal `0`.
+
+The product of `tupleSize` and `numberOfTuples` MUST be less than or equal to the number of characters in the hex encoded digest.
+
+#### numberOfTuples
+
+`numberOfTuples` determines how many tuples to create from the digest. The tuples are used as directory names, and each successive directory is nested within the previous. The default value is `3`, which means that every OCFL object root will be 4 directories removed from the OCFL storage root, 3 tuple directories plus 1 encapsulation directory. Increasing this value increases the depth of the OCFL storage hierarchy.
+
+If `numberOfTuples` is set to `0`, then no tuples are created and `tupleSize` MUST also equal `0`.
+
+The product of `numberOfTuples` and `tupleSize` MUST be less than or equal to the number of characters in the hex encoded digest.
+
+## Procedure
+
+The following is an outline of the steps to follow to map an OCFL object identifier to an OCFL object root path using this extension (also see the Python Code section):
+
+1. The OCFL object identifier is encoded as UTF-8 and hashed using the specified `digestAlgorithm`.
+2. The digest is encoded as a lower-case hex string.
+3. Starting at the beginning of the digest and working forwards, the digest is divided into `numberOfTuples` tuples each containing `tupleSize` characters.
+4. The tuples are joined, in order, using the filesystem path separator.
+5. The OCFL object identifier is percent-encoded to create the encapsulation directory name (see ["Encapsulation Directory"](#encapsulation-directory) section above for details).
+6. The encapsulation directory name is joined to the end of the path.
+
+## Examples
+
+### Example 1
+
+This example demonstrates what the OCFL storage hierarchy looks like when using this extension's default configuration.
+
+#### Parameters
+
+It is not necessary to specify any parameters to use the default configuration. However, if you were to do so, it would look like the following:
+
+```json
+{
+    "extensionName": "NNNN-hash-and-id-n-tuple-storage-layout",
+    "digestAlgorithm": "sha256",
+    "tupleSize": 3,
+    "numberOfTuples": 3
+}
+```
+
+#### Mappings
+
+| Object ID | Digest | Object Root Path |
+| --- | --- | --- |
+| object-01 | 3c0ff4240c1e116dba14c7627f2319b58aa3d77606d0d90dfc6161608ac987d4 | `3c0/ff4/240/object-01` |
+| ..hor/rib:le-$id | 487326d8c2a3c0b885e23da1469b4d6671fd4e76978924b4443e9e3c316cda6d | `487/326/d8c/%2e%2ehor%2frib%3ale-%24id` |
+
+#### Storage Hierarchy
+
+```
+[storage_root]/
+├── 0=ocfl_1.0
+├── ocfl_layout.json
+├── NNNN-hashed-n-tuple-id-layout.json
```

Comment:
This file should now be at `extensions/hash-and-id-n-tuple-storage-layout/config.json`